runtime/pkg/graceful/postgres.go

```diff
@@ -0,0 +1,65 @@
+package graceful
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net"
+	"strings"
+
+	wire "github.com/jeroenrinzema/psql-wire"
+	"go.uber.org/zap"
+)
+
+// ServePostgres serves a Postgres server and performs a graceful shutdown if/when ctx is cancelled.
+func ServePostgres(ctx context.Context, queryHandler func(ctx context.Context, query string) (wire.PreparedStatements, error), authHandler func(ctx context.Context, username, password string) (context.Context, bool, error), port int, useTLS bool, logger *zap.Logger) error {
```

Comment:
3. Yeah I was able to setup TLS termination as well but did not implement it because nginx would normally terminate TLS connections. 
I also see that nginx plus is able to terminate SSL for TCP connections as well : https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-tcp/
Do you mean it is only supported for nginx plus and not for community version ?
Though I see that directives are present for community version as well : https://nginx.org/en/docs/stream/ngx_stream_ssl_module.html?&_ga=2.117547401.102737501.1716965242-1274175600.1685512674#example

Reply:
What I mean is that it seems Postgres implements TLS inside their protocol, not on the TCP connection itself: https://stackoverflow.com/questions/45542830/nginx-tls-termination-for-postgresql